dot/network/README.md

```diff
@@ -0,0 +1,178 @@
+Kademlia as a protocol - in particular, these three properties are:
+
+- the "distance" between a peer and itself is zero
+- the "distance" between two peers is the same regardless of the order in which the peers are considered (it is
+  [symmetric](https://en.wikipedia.org/wiki/Symmetry_in_mathematics))
+- the shortest "distance" between two peers does not include any intermediate peers (it follows the
+  [triangle inequality](https://en.wikipedia.org/wiki/Triangle_inequality))
+
+Gossamer uses [the official `libp2p` implementation of Kademlia for Go](https://github.com/libp2p/go-libp2p-kad-dht).
+
+### Blockchain-Specific Protocols
+
+The `libp2p` stack is used to implement the blockchain-specific protocols that are used to participate in
+"Substrate-like" networks - these protocols are divided into two types, [notification](#notification-protocols) and
+[request/response](#requestresponse-protocols). The two types of protocols are described in greater details below, along
+with the specific protocols for each type.
+
+##### Notification Protocols
+
+[Notification protocols](https://crates.parity.io/sc_network/index.html#notifications-protocols) allow peers to
+unidirectionally "push" information to other peers in the network. Although the peer receiving the information must
+explicitly accept a handshake in order to open a stream for a notification protocol, this stream does not allow the
+receiver to "push" information to the sender.
+
+###### Transactions
+
+This protocol is used to notify network peers of [transactions](https://docs.substrate.io/v3/concepts/tx-pool/) that
+have been locally received and validated. Transactions are used to access the
+[public APIs of blockchain runtimes](https://docs.substrate.io/v3/concepts/extrinsics/#signed-transactions).
+
+###### Block Announces
+
+The block announce protocol is used to notify network peers of the existence of a block. The message for this protocol
+contains a [block header](https://docs.substrate.io/v3/getting-started/glossary/#header) and associated data, such as
+the [BABE pre-runtime digest](https://crates.parity.io/sp_consensus_babe/digests/enum.PreDigest.html).
+
+###### GRANDPA
+
+[Finality](https://wiki.polkadot.network/docs/learn-consensus#finality-gadget-grandpa) protocols ("gadgets") such as
+GRANDPA are often described in terms of "games" that are played by the participants in a network. In GRANDPA, this game
+relates to voting on the blocks that inform the canonical chain. This notification protocol is used by peers to cast
+votes for participation in the GRANDPA game.
+
+##### Request/Response Protocols
+
+[These protocols](https://crates.parity.io/sc_network/index.html#request-response-protocols) allow peers to request
+specific information from one another. The requesting peer sends a protocol-specific message that describes the request
+and the peer to which the request was sent replies with a message.
```

Comment:
I'd update this to also say if a node opens a stream with some remote peer, then the outgoing side will *only* push requests and the incoming side will only expect to receive requests. same for the other way around, the outgoing side expects to only receive responses.